src/rpc/rawtransaction.cpp

```diff
@@ -304,7 +304,7 @@ static RPCHelpMan createrawtransaction()
         }, true
     );
 
-    bool rbf = false;
+    std::optional<bool> rbf;
     if (!request.params[3].isNull()) {
         rbf = request.params[3].isTrue();
     }
```

Comment:
```suggestion
    const bool rbf{request.params[3].isNull() ? true : request.params[3].getBool()};
```

Seems complicated to use `optional<bool>` when it is not needed?

Reply:
The use of the optional was done to allow for the current behavior with user defined sequence numbers to work. There is a check in `ConstructTransaction` that the sequence numbers match the `replaceable` option. It will throw an error if non-RBF sequences are used, but replaceable is true. Additionally, if `replacable` is either not specified or false, setting a RBF sequence does not result in an error. If the default is just changed to `true`, then this check fails when the user sets non-RBF sequence numbers and does not specify `replaceable` at all. By using an optional we can retain the behavior that sequences are not checked against `replaceable` when `replaceable` is not specified.